web/src/main/java/org/wso2/testgrid/web/api/APIUtil.java

```diff
@@ -117,21 +121,23 @@ static DeploymentPattern getDeploymentPatternBean(org.wso2.testgrid.common.Deplo
     /**
      * Util method to convert {@link org.wso2.testgrid.common.TestPlan} instance to a {@link TestPlan} instance.
      *
-     * @param testPlan {@link org.wso2.testgrid.common.TestPlan} instance to be converted
+     * @param testPlan             {@link org.wso2.testgrid.common.TestPlan} instance to be converted
      * @param requireTestScenarios boolean flag to indicate whether to include test-scenario info
      * @return {@link TestPlan} instance  with necessary information
      */
     static TestPlan getTestPlanBean(org.wso2.testgrid.common.TestPlan testPlan, boolean requireTestScenarios) {
         TestPlan testPlan1 = new TestPlan();
-        testPlan1.setId(testPlan.getId());
-        testPlan1.setDeploymentPattern(testPlan.getDeploymentPattern().getName());
-        testPlan1.setDeploymentPatternId(testPlan.getDeploymentPattern().getId());
-        testPlan1.setInfraParams(testPlan.getInfraParameters());
-        testPlan1.setStatus(testPlan.getStatus().toString());
-        testPlan1.setCreatedTimestamp(testPlan.getCreatedTimestamp());
-        testPlan1.setModifiedTimestamp(testPlan.getModifiedTimestamp());
-        if (requireTestScenarios) {
-            testPlan1.setTestScenarios(getTestScenarioBeans(testPlan.getTestScenarios(), false));
+        if (testPlan != null) {
+            testPlan1.setId(testPlan.getId());
```

Comment:
Shall we change the variable names `testPlan` and `testPlan1` to something more meaningful since it is hard to distinguish which one is which at the moment